scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,482 @@
+    year_range = str(max_year) + '-' + str(min_year)
+    return year_range
+
+
+def _clean_csv_file(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    This method cleans the dataframe loaded from a csv file format.
+    Also, Performs transformations on the data.
+
+    Arguments:
+        df (DataFrame) : DataFrame of csv dataset
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for txt dataset.
+    """
+    idx = df[df[0] == "Year and Month"].index
+    df = df.iloc[idx.values[0] + 1:][:]
+    df = df.dropna(axis=1, how='all')
+    cols = [
+        "Year and Month", "Resident Population",
+        "Resident Population Plus Armed Forces Overseas", "Civilian Population",
+        "Civilian NonInstitutionalized Population"
+    ]
+    df.columns = cols
+    for col in df.columns:
+        df[col] = df[col].str.replace(",", "")
+    return df
+
+
+class CensusUSACountryPopulation:
+    """
+    CensusUSACountryPopulation class provides methods
+    to load the data into dataframes, process, cleans
+    dataframes and finally creates single cleaned csv
+    file.
+    Also provides methods to generate MCF and TMCF
+    Files using pre-defined templates.
+    """
+
+    def __init__(self, input_files: list, csv_file_path: str,
+                 mcf_file_path: str, tmcf_file_path: str) -> None:
+        self.input_files = input_files
+        self.cleaned_csv_file_path = csv_file_path
+        self.mcf_file_path = mcf_file_path
+        self.tmcf_file_path = tmcf_file_path
+        self.df = None
+        self.file_name = None
+        self.scaling_factor = 1
+
+    def _load_data(self, file: str) -> pd.DataFrame:
+        """
+        This Methods loads the data into pandas Dataframe
+        using the provided file path and Returns the Dataframe.
+
+        Arguments:
+            file (str) : String of Dataset File Path
+        Returns:
+            df (DataFrame) : DataFrame with loaded dataset
+        """
+        df = None
+        self.file_name = os.path.basename(file)
+        if ".xls" in file:
+            df = pd.read_excel(file)
+        elif ".csv" in file:
+            file_name = file_name.replace(".csv", ".xlsx")
+            df = pd.read_csv(file, header=None)
+            df = _clean_csv_file(df)
+
+        elif ".txt" in file:
+            skip_rows_txt = 17
+            self.file_name = self.file_name.replace(".txt", ".xlsx")
+            cols = [
+                "Year and Month", "Date", "Resident Population",
+                "Resident Population Plus Armed Forces Overseas",
+                "Civilian Population",
+                "Civilian NonInstitutionalized Population"
+            ]
+            df = pd.read_table(file,
+                               index_col=False,
+                               delim_whitespace=True,
+                               engine='python',
+                               skiprows=skip_rows_txt,
+                               names=cols)

```

Comment:
Haven't you don a bunch of this cleaning already in download.py?

Reply:
Extra/Redundant lines of code have been removed from preprocess.py